target/linux/mediatek/mt7622/base-files/etc/board.d/02_network

```diff
@@ -22,16 +22,40 @@ mediatek_setup_interfaces()
 	ubnt,unifi-6-lr-ubootmod)
 		ucidef_set_interface_lan "eth0"
 		;;
+	xiaomi,redmi-router-ax6s)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" wan
+		;;
 	*)
 		ucidef_add_switch "switch0" \
 				  "0:lan" "1:lan" "2:lan" "3:lan" "4:wan" "6u@eth0" "5u@eth1"
 		;;
 	esac
 }
 
+mediatek_setup_macs()
+{
+	local board="$1"
+	local lan_mac=""
+	local wan_mac=""
+	local label_mac=""
+
+	case $board in
+	xiaomi,xiaomi-redmi-router-ax6s)
+		lan_mac=$(mtd_get_mac_ascii u-boot-env ethaddr)
```

Comment:
Doesn't the bootloader pass down `ethaddr` to the kernel anyway to be applied for the first Ethernet interface? Ie. no need to handle `lan_mac` here (unless stock U-Boot is really weird on that board)

Reply:
Ah, I see. Aliasing ethernet0 to gmac0 gets that done. ethaddr_wan might have to be done here still though?

Reply:
Yes, `wan` address will need to be set here.